<launch>

    <arg name="use_gripper" default="false"/>
    <arg name="record" default="false"/>
    <arg if="$(arg record)" name="prefix"/>

    <node pkg="intel_aero_ros" type="landing_node" name="landing_node" output="screen">
        <remap from="state" to="/mavros/state"/>
        <remap from="pose" to="/mavros/local_position/pose"/>
        <remap from="target" to="/mavros/setpoint_raw/local"/>
        <remap from="lengths" to="/gripper/lengths"/>
        <remap from="rc_override" to="/mavros/rc/override"/>
        <param name="dist_threshold" value="0.3"/>
        <param name="hover_duration" value="6.0"/>
        <param name="start_settle_duration" value="10.0"/>
        <param name="land_duration" value="1.0"/>
        <rosparam param="start_position">[-1.0, -1.5, 2.5]</rosparam>
        <rosparam param="land_velocity">[0.0, 0.0, -0.1]</rosparam>
        <rosparam param="lengths">[121, 121, 138, 138]</rosparam>
    </node>

    <node if="$(arg use_gripper)" pkg="intel_aero_ros" type="gripper_node" name="gripper" output="screen">
        <remap from="lengths" to="/gripper/lengths"/>
    </node>

    <node if="$(arg record)" pkg="rosbag" type="record" name="recorder" args="-a -o $(arg prefix)"/>

</launch>
